kitsune/search/v2/documents.py

```diff
@@ -111,47 +173,46 @@ class Index:
         using = config.DEFAULT_ES7_CONNECTION
 
     @classmethod
-    def prepare_locale(cls, instance):
-        return cls.get_question_instance(instance).locale
+    def prepare_question_creator(cls, instance):
+        return UserInnerDoc.prepare(instance.creator)
 
     @classmethod
-    def prepare(cls, instance):
-        """Prepare an object given a model instance"""
+    def prepare_question_updated_by(cls, instance):
+        return UserInnerDoc.prepare(instance.updated_by)
 
-        fields = ["question_id", "question_title", "question_content", "locale"] + cls.get_fields()
-        locale_fields = ["question_content"]
+    @classmethod
+    def prepare_question_marked_as_spam_by(cls, instance):
+        return UserInnerDoc.prepare(instance.marked_as_spam_by)
 
-        obj = cls()
+    @classmethod
+    def prepare_question_product(cls, instance):
+        return ProductInnerDoc.prepare(instance.product)
 
-        # Iterate over fields and either set the value directly from the instance
-        # or prepare based on `prepare_<field>` method
-        for f in fields:
-            try:
-                prepare_method = getattr(obj, "prepare_{}".format(f))
-                value = prepare_method(instance)
-            except AttributeError:
-                if f.startswith("question_"):
-                    value = getattr(cls.get_question_instance(instance), f[len("question_") :])
-                else:
-                    value = getattr(instance, f)
+    @classmethod
+    def prepare_question_topic(cls, instance):
+        return TopicInnerDoc.prepare(instance.topic)
 
-            if f in locale_fields:
-                setattr(obj, f, {})
-                setattr(obj[f], cls.prepare_locale(instance), value)
-            else:
-                setattr(obj, f, value)
+    @classmethod
+    def prepare_question_taken_by(cls, instance):
+        return UserInnerDoc.prepare(instance.taken_by)
 
-        obj.meta.id = instance.id
+    @classmethod
+    def prepare_question_tags(cls, instance):
+        return [TagInnerDoc.prepare(tag) for tag in instance.tags.all()]
 
-        return obj
+    @classmethod
+    def prepare_question_has_solution(cls, instance):
+        return instance.solution_id is not None
 
     @classmethod
-    def get_fields(cls):
-        return []
+    def prepare_locale(cls, instance):
```

Comment:
this already exists in the parent class and it can be removed from here since it's the same.